test/middleware/express.test.ts

```diff
@@ -0,0 +1,341 @@
+                        const next: NextFunction = jest.fn();
+                        middleware(complexRequest as Request, mockResponse as Response, next);
+                        expect(next).toBeCalledTimes(1);
+                        expect(next).toBeCalledWith();
+
+                        // advance the timers by 1 second for the next request
+                        jest.advanceTimersByTime(1000);
+                    }
+                });
+
+                test('Multiple valid requests at within one second', () => {
+                    for (let i = 0; i < 3; i++) {
+                        const next: NextFunction = jest.fn();
+                        middleware(complexRequest as Request, mockResponse as Response, next);
+                        expect(next).toBeCalledTimes(1);
+                        expect(next).toBeCalledWith();
+
+                        // advance the timers by 1 second for the next request
+                        jest.advanceTimersByTime(20);
+                    }
+                });
+            });
+
+            describe('BLOCKS requests', () => {
+                test('A single request that exceeds capacity', () => {
+                    complexRequest = {
+                        // complexity should be 12 if 'first' is accounted for.
+                        // scalars: 1, droid: 1, reviews (5 * (1 Review, 1 episode))
+                        query: {
+                            query: `Query {
+                            scalars: {
+                                num
+                            }
+                            droid(id: 1) {
+                                name
+                            }
+                            reviews(episode: 'NEWHOPE', first: 5) {
+                                episode 
+                                stars
+                                commentary
+                            }
+                        `,
+                        },
+                    };
+                    const expectedResponse = {
+                        status: 429,
+                    };
+
+                    middleware(mockRequest as Request, mockResponse as Response, nextFunction);
+                    expect(mockResponse.status).toBe(429);
+                    expect(nextFunction).not.toBeCalled();
+
+                    // FIXME: There are multiple functions to send a response
+                    // json, send html, sendStatus etc. How do we check at least one was called
+                    expect(mockResponse.send).toBeCalled();
+                });
+
+                test('Multiple queries that exceed token limit', () => {
+                    for (let i = 0; i < 5; i++) {
+                        // Send 5 queries of complexity 2. These should all succeed
+                        middleware(mockRequest as Request, mockResponse as Response, nextFunction);
+
+                        // advance the timers by 1 second for the next request
```

Comment:
The comment says `1 second` and the code says `20ms`. I think the code is right but the intention is muddled.